tools/build-webview

```diff
@@ -29,95 +37,126 @@ err() {
     exit 1;
 }
 
+# Common rsync invocation. Exported for use by auxiliary scripts.
+#
+# Filter rules are expected to be passed in via stdin; this is most easily
+# accomplished with a heredoc, or `</dev/null` if unneeded.
+#
+# The last argument supplied must be the destination directory. (This wrapper
+# does not support mixing positional and named arguments; positionals must come
+# last.)
+z_rsync() {
+    # Create the destination directory. (rsync won't generally do this.)
+    mkdir -p "${@: -1}"
+    # Invoke rsync with the standard rules and exclusions, as well as whatever
+    # other rules were specified by our invoker.
+    rsync -a --no-D --delete --delete-excluded \
+        --exclude=README.md --exclude=.gitignore \
+        --filter='. /dev/stdin' \
+        "$@"
+}
+export -f z_rsync
+
 ################################################################################
 # Parameters and environment
 ################################################################################
 
-# chdir to the current git repo's root
-ROOT="$(git rev-parse --show-toplevel)"
-cd "$ROOT"
+# Chdir to the current git repo's root.
+readonly _root="$(git rev-parse --show-toplevel)"
+cd "$_root"
```

Comment:
While we were discussing in person today, you noticed that this `cd` should come after the `readlink -m` below, in order to treat the destination argument in the standard CLI way if the script is run from a cwd that happens not to be the repo root.